### DISCUSSION TITLE
About 404 page

### DISCUSSION THREAD

**Discussion by tffaruk** [vercel/next.js#49894] [App Router] [answered]
There is a 500 internal server error  instead of 404 page after deploying on vercel.
because a dynamic route and not-found.tsx page in the same directory in app route
1. How to solve this problem?
2. Is this possible that not-found and dynamic route in the same directory?

**Comment by devjiwonchoi** [comment, marked as answer]
If the not-found.tsx is not in the root directory, it is recommended to call the notFound() function inside the [id]/page.tsx.

Example: 
```
import { notFound } from 'next/navigation';
 
async function fetchUsers(id) {
  const res = await fetch('https://...');
  if (!res.ok) return undefined;
  return res.json();
}
 
export default async function Profile({ params }) {
  const user = await fetchUser(params.id);
 
  if (!user) {
    notFound();
  }
 
  // ...
}
```
ref: https://nextjs.org/docs/app/api-reference/functions/not-found

I prefer locating not-found.tsx on the root directory for coherence of the project.

For the second question, yes it is possible.
